test/extensions/retry_policy/conditional_retry/config_test.cc

```diff
@@ -0,0 +1,245 @@
+#include "extensions/retry_policy/conditional_retry/config.h"
+#include "envoy/registry/registry.h"
+
+#include "test/test_common/utility.h"
+
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace RetryPolicy {
+namespace ConditionalRetry {
+
+TEST(ConfigFactoryTest, TestFactory) {
+  auto factory = Registry::FactoryRegistry<Router::RetryPolicyFactory>::getFactory(
+      RetryPolicyValues::get().ConditionalRetry);
+  EXPECT_EQ(factory->name(), RetryPolicyValues::get().ConditionalRetry);
+
+  const std::string yaml = R"EOF(
+      retry_conditions:
+      - request_header:
+          name: ":method"
+          exact_match: "GET"
+        retry_on: "connect-failure"
+)EOF";
+  ProtobufTypes::MessagePtr proto_config = factory->createEmptyConfigProto();
+  TestUtility::loadFromYaml(yaml, *proto_config);
+
+  auto policy =
+      factory->createRetryPolicy(*proto_config, Http::TestRequestHeaderMapImpl{{":method", "GET"}},
+                                 Envoy::ProtobufMessage::getStrictValidationVisitor());
+  EXPECT_NE(nullptr, policy);
+}
+
+TEST(ConfigFactoryTest, TestConditionalRetry) {
```

Comment:
@mattklein123 added an extension example. I am going to add the doc for the extension. This test case shows how to configure the extension.